src/web_clipboard.rs

```diff
@@ -0,0 +1,146 @@
+use crossbeam_channel::{Receiver, Sender};
+
+use bevy::prelude::*;
+use wasm_bindgen_futures::spawn_local;
+
+use crate::EguiClipboard;
+use wasm_bindgen::closure::Closure;
+use wasm_bindgen::prelude::*;
+
+/// startup system for bevy to initialize web events.
+pub fn startup_setup_web_events(mut clipboard_channel: ResMut<EguiClipboard>) {
+    setup_clipboard_copy(&mut clipboard_channel.web_copy);
+    setup_clipboard_cut(&mut clipboard_channel.web_cut);
+    setup_clipboard_paste(&mut clipboard_channel.web_paste);
+}
+
+/// To get data from web events
+#[derive(Default)]
+pub struct WebChannel<T> {
+    rx: Option<Receiver<T>>,
+}
+
+impl<T> WebChannel<T> {
+    /// Only returns Some if user explicitly triggered an event. Should be called each frame to react as soon as the event is fired.
+    pub fn try_read_clipboard_event(&mut self) -> Option<T> {
+        match &mut self.rx {
+            Some(rx) => {
+                if let Ok(data) = rx.try_recv() {
+                    return Some(data);
+                }
+                None
+            }
+            None => None,
+        }
+    }
+}
+
+/// User provided a string to paste
+#[derive(Debug, Default)]
+pub struct WebEventPaste(pub String);
+/// User asked to cut
+#[derive(Default)]
+pub struct WebEventCut;
+/// Used asked to copy
+#[derive(Default)]
+pub struct WebEventCopy;
+
+fn setup_clipboard_copy(clipboard_channel: &mut WebChannel<WebEventCopy>) {
+    let (tx, rx): (Sender<WebEventCopy>, Receiver<WebEventCopy>) = crossbeam_channel::bounded(1);
+
+    let closure = Closure::<dyn FnMut(_)>::new(move |event: web_sys::ClipboardEvent| {
+        // TODO: maybe we should check if current canvas is selected ? not sure it's possible,
+        // but reacting to event at the document level will lead to problems if multiple games are on the same page.
+        tx.try_send(WebEventCopy);
+    });
+
+    // TODO: a lot of unwraps ; it's using documents because paste event set on a canvas do not trigger (also tested on firefox in vanilla javascript)
```

Comment:
any feeling about this ? should it fail silently ? with a warning/error log?

Reply:
I'd change `unwrap` to `expect`, otherwise seems fine to me, as panicking on initialization failure doesn't seem to be silent :)